presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java

```diff
@@ -62,6 +63,21 @@ private OperatorAssertion()
 
     public static List<Page> toPages(Operator operator, Iterator<Page> input)
     {
+        try {
+            return toPagesUnsafe(operator, input);
+        }
+        finally {
+            try {
+                operator.close();
```

Comment:
I think a better alternative would be to make `toPages` take an `OperatorFactory` and `DriverContext` instead of `Operator`. This way, it would make sense to close the `Operator` in this method because that's where you created it.

I checked all the callers of `toPages(Operator operator, Iterator<Page> input)`. All but one (`TestHashAggregationOperator`) should have used `toPages(Operator operator, List<Page> input)`. And only `TestHashAggregationOperator` doesn't work with the interface I proposed.

So, I would migrate all callers of `toPages(Operator operator, Iterator<Page> input)` and `toPages(Operator operator, List<Page> input)` to the new method (except `TestHashAggregationOperator`). For `TestHashAggregationOperator`, I will just add a try-with-resource to `testMultiplePartialFlushes` method when the operator is created.